aten/src/ATen/test/test_parallel.cpp

```diff
@@ -9,12 +9,24 @@
 #include <string.h>
 #include <sstream>
 
+struct NumThreadsGuard {
+  int old_num_threads_;
+  NumThreadsGuard(int nthreads) {
+    old_num_threads_ = at::get_num_threads();
+    at::set_num_threads(nthreads);
+  }
+
+  ~NumThreadsGuard() {
+    at::set_num_threads(old_num_threads_);
+  }
+};
+
 using namespace at;
 
 // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
 TEST(TestParallel, TestParallel) {
   manual_seed(123);
-  set_num_threads(1);
+  NumThreadsGuard guard(1);
```

Comment:
This test was having the global side-effect of disabling multi-threading.

Reply:
Thanks for fixing this!